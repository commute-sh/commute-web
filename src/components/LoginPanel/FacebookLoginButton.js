import React, { Component, PropTypes } from 'react';

import RaisedButton from 'material-ui/RaisedButton';

import Icon from './Icon';

class FacebookLoginButton extends Component {

  static propTypes = {
    url: PropTypes.string
  };

  static defaultProps = {
    url: '/auth/facebook'
  };

  render() {
    return (
      <RaisedButton
        href={this.props.url}
        label="Facebook"
        backgroundColor="#3b5998"
        labelColor="white"
        fullWidth={true}
        style={Object.assign({ height: 44 }, this.props.style)}
        icon={<Icon name='facebook' />}
      />
    );
  }

}

export default FacebookLoginButton;
