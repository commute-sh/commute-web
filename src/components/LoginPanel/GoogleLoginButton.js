import React, { Component, PropTypes } from 'react';

import RaisedButton from 'material-ui/RaisedButton';

import Icon from './Icon';

class GoogleLoginButton extends Component {

  static propTypes = {
    url: PropTypes.string
  };

  static defaultProps = {
    url: '/auth/google'
  };

  render() {
    return (
      <RaisedButton
        href={this.props.url}
        label="Google"
        backgroundColor="#dd4b39"
        labelColor="white"
        fullWidth={true}
        style={Object.assign({ height: 44 }, this.props.style)}
        icon={<Icon name='google' />}
      />
    );
  }

}

export default GoogleLoginButton;
