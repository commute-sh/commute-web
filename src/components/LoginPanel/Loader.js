import React, { Component, PropTypes } from 'react'
import CircularProgress from 'material-ui/CircularProgress';

class Loader extends Component {

  static propTypes = {
    style: PropTypes.object
  };

  static defaultProps = {
    style: {
      background: '#fff',
      opacity: 0.8
    }
  };

  render() {
    return (
      <div style={[{ position: 'relative' }, this.props.style]}>
        <div style={{
          zIndex: 5,
          background: '#fff', opacity: 0.8,
          position: 'absolute', top: 0, bottom: 0, left: 0, right: 0,
          display: 'flex', alignItems: 'center', justifyContent: 'center'
        }}>
          <CircularProgress />
        </div>
      </div>
    )
  }

}

export default Loader;
