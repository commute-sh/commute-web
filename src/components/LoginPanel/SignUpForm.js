import React, { Component, PropTypes } from 'react'
import { reduxForm, Field } from 'redux-form'
import { TextField } from 'redux-form-material-ui'
import { RaisedButton } from 'material-ui'

const validate = values => {
  const errors = {};
  const requiredFields = [ 'email', 'password' ];

  requiredFields.forEach(field => {
    if (!values[ field ]) {
      errors[ field ] = 'Required'
    }
  });

  if (values.email && !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
    errors.email = 'Invalid email address'
  }

  return errors;
};


class SignUpForm extends Component {

  static propTypes = {
    handleSignUp: PropTypes.func
  };

  componentDidMount() {
    this.refs.email            // the Field
      .getRenderedComponent() // on Field, returns ReduxFormMaterialUITextField
      .getRenderedComponent() // on ReduxFormMaterialUITextField, returns TextField
      .focus();                // on TextField
  }

  render() {

    const { handleSubmit, handleSignUp, pristine, submitting } = this.props;

    return (
      <form onSubmit={handleSubmit(handleSignUp)}>
        <Field name="email" component={TextField}
               floatingLabelText="Email"
               fullWidth={true}
               type="email"
               floatingLabelFixed={true}
               inputStyle={{ marginLeft: 4, marginTop: 4, fontSize: '14px' }}
               floatingLabelStyle={{ top: 24, fontFamily: 'Lobster' }}
               floatingLabelFocusStyle={{ top: 24, fontFamily: 'Lobster' }}
               style={{ height: 58 }}

               ref="email" withRef />

        <Field name="password" component={TextField}
               floatingLabelText="Mot de passe"
               floatingLabelFixed={true}
               fullWidth={true}
               type="password"
               inputStyle={{ marginLeft: 4, marginTop: 10, fontSize: '14px' }}
               floatingLabelStyle={{ top: 34, fontFamily: 'Lobster' }}
               floatingLabelFocusStyle={{ top: 34, fontFamily: 'Lobster' }}
        />

        <RaisedButton
          type="submit"
          label="CrÃ©er votre compte"
          disabled={pristine || submitting}
          backgroundColor="#345d79"
          labelColor="white"
          fullWidth={true}
          labelStyle={{ fontFamily: 'Lobster', textTransform: 'none' }}
          style={{ marginTop: 10, height: 44 }}
        />

      </form>
    )
  }

}

SignUpForm = reduxForm({
  form: 'signUpForm',
  initialValues: {
    email: '',
    password: ''
  },
  validate
})(SignUpForm);

export default SignUpForm;
