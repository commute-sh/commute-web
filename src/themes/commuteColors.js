'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var commute50 = exports.commute50 = '#cbdeea';
var commute100 = exports.commute100 = '#95bad4';
var commute200 = exports.commute200 = '#6da1c3';
var commute300 = exports.commute300 = '#447ea5';
var commute400 = exports.commute400 = '#3b6e90';
var commute500 = exports.commute500 = '#325d7a';
var commute600 = exports.commute600 = '#294c64';
var commute700 = exports.commute700 = '#203c4f';
var commute800 = exports.commute800 = '#172b39';
var commute900 = exports.commute900 = '#0e1b23';
var commuteA100 = exports.commuteA100 = '#cbdeea';
var commuteA200 = exports.commuteA200 = '#95bad4';
var commuteA400 = exports.commuteA400 = '#3b6e90';
var commuteA700 = exports.commuteA700 = '#203c4f';
