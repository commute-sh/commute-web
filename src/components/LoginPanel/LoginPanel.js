import React, { Component, PropTypes } from 'react'

import FacebookLoginButton from './FacebookLoginButton'
import GoogleLoginButton from './GoogleLoginButton'
import TwitterLoginButton from './TwitterLoginButton'
import Paper from 'material-ui/Paper'
import TextField from 'material-ui/TextField';
import RaisedButton from 'material-ui/RaisedButton';

class LoginPanel extends Component {

  render() {
    return (

      <div>

        <div style={{ fontFamily: 'Lobster', color: '#ccc', textAlign: 'center', padding: 20, fontSize: 20 }}>Se connecter</div>

        <div style={{ backgroundColor: 'white', display: 'flex', flex: '1', flexDirection: 'row', width: 640, padding: 12 }}>

          <div style={{ flex: 1 }}>
            <div style={{ width: '100%', padding: 6 }}>
              <FacebookLoginButton />
            </div>
            <div style={{ width: '100%', padding: 6 }}>
              <GoogleLoginButton />
            </div>
            <div style={{ width: '100%', padding: 6 }}>
              <TwitterLoginButton />
            </div>
          </div>

          <div style={{ textAlign: 'center', margin: '20px', width: 1, backgroundColor: '#e0e0e0' }}>
          </div>

          <Paper zDepth={0} style={{ textAlign: 'center', padding: 6, flex: 1 }}>

            <TextField
              floatingLabelText="Email"
              fullWidth={true}
              floatingLabelFixed={true}
              inputStyle={{ marginLeft: 4, marginTop: 4, fontSize: '14px' }}
              floatingLabelStyle={{ top: 24, fontFamily: 'Lobster' }}
              floatingLabelFocusStyle={{ top: 24, fontFamily: 'Lobster' }}
              style={{ height: 58 }}
            />

            <TextField
              floatingLabelText="Mot de passe"
              floatingLabelFixed={true}
              fullWidth={true}
              type="password"
              inputStyle={{ marginLeft: 4, marginTop: 10, fontSize: '14px' }}
              floatingLabelStyle={{ top: 34, fontFamily: 'Lobster' }}
              floatingLabelFocusStyle={{ top: 34, fontFamily: 'Lobster' }}
            />

            <RaisedButton
              href="/auth/login"
              label="Se connecter"
              backgroundColor="#345d79"
              labelColor="white"
              fullWidth={true}
              labelStyle={{ fontFamily: 'Lobster', textTransform: 'none' }}
              style={{ marginTop: 10, height: 44 }}
            />

            <RaisedButton
              href="/auth/sign-up"
              label="Sign Up"
              backgroundColor="#f39c12"
              labelColor="#222"
              fullWidth={true}
              labelStyle={{ fontFamily: 'Lobster', textTransform: 'none' }}
              style={{ marginTop: 10, height: 44 }}
            />

          </Paper>

        </div>
      </div>

    );
  }

}

export default LoginPanel;
