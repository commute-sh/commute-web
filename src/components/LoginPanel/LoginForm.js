import React, { Component, PropTypes } from 'react'
import { reduxForm, Field } from 'redux-form'
import { TextField } from 'redux-form-material-ui'
import { RaisedButton } from 'material-ui'
import Loader from './Loader';

const validate = values => {
  const errors = {};
  const requiredFields = [ 'username', 'password' ];

  requiredFields.forEach(field => {
    if (!values[ field ]) {
      errors[ field ] = 'Required'
    }
  });

  return errors;
};

class LoginForm extends Component {

  static propTypes = {
    submitTitle: PropTypes.string,
    login: PropTypes.shape({
      isFetching: PropTypes.bool,
      errMessage: PropTypes.string
    }),
    onLoginSubmit: PropTypes.func
  };

  render() {

    const { submitTitle, handleSubmit, onLoginSubmit, pristine, submitting, invalid, login: { isFetching, errMessage } } = this.props;

    return (
      <div style={{ position: 'relative' }}>

        { isFetching &&
          <Loader style={{
            zIndex: 5,
            position: 'absolute', top: 0, bottom: 0, left: 0, right: 0
          }} />
        }

        { errMessage &&
            <div style={{ color: 'red', fontSize: 12, padding: 5, paddingBottom: 10 }}>
              {errMessage}
            </div>
        }

        <form onSubmit={handleSubmit(onLoginSubmit)}>
          <Field name="username" component={TextField}
                 floatingLabelText="Identifiant"
                 fullWidth={true}
                 type="username"
                 floatingLabelFixed={true}
                 inputStyle={{ marginLeft: 4, marginTop: 4, fontSize: '14px' }}
                 floatingLabelStyle={{ top: 24, fontFamily: 'Lobster' }}
                 floatingLabelFocusStyle={{ top: 24, fontFamily: 'Lobster' }}
                 style={{ height: 58 }}
                 errorStyle={{ marginTop: 6, textAlign: 'left' }}
                 ref="username" withRef />

          <Field name="password" component={TextField}
                 floatingLabelText="Mot de passe"
                 floatingLabelFixed={true}
                 fullWidth={true}
                 type="password"
                 inputStyle={{ marginLeft: 4, marginTop: 10, fontSize: '14px' }}
                 floatingLabelStyle={{ top: 34, fontFamily: 'Lobster' }}
                 floatingLabelFocusStyle={{ top: 34, fontFamily: 'Lobster' }}
                 errorStyle={{ textAlign: 'left' }}
          />

          <RaisedButton
            type="submit"
            label={submitTitle}
            disabled={pristine || submitting || invalid}
            backgroundColor="#345d79"
            labelColor="white"
            fullWidth={true}
            labelStyle={{ fontFamily: 'Lobster', textTransform: 'none' }}
            style={{ marginTop: 10, height: 44 }}
          />

        </form>
      </div>
    )
  }

}

export default reduxForm({
  form: 'loginForm',
  initialValues: {
    email: '',
    password: ''
  },
  validate
})(LoginForm);
