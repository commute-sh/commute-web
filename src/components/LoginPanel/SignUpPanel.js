import React, { Component, PropTypes } from 'react'

import Paper from 'material-ui/Paper'
import SignUpForm from './SignUpForm'


class SignUpPanel extends Component {

  static propTypes = {
    handleSignUp: PropTypes.func
  };

  render() {

    return (
      <Paper zDepth={0} style={{ textAlign: 'center', padding: 6, flex: 1 }}>
        <SignUpForm handleSignUp={this.props.handleSignUp} />
      </Paper>
    );

  }

}

export default SignUpPanel;
