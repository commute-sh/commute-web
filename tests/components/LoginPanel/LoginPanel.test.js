// Link.react-test.js
import React from 'react';
import LoginPanel from '../../../src/components/LoginPanel';
import renderer from 'react-test-renderer';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import { commute500, commute700 } from '../../../src/themes/commuteColors';


////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/// Theme
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

const commuteTheme = getMuiTheme({
  palette: {
    primary1Color: commute500,
    primary2Color: commute700,
  },
  appBar: {
    height: 64,
  },
});


////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
/// Tests
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

test('LoginPanel renders correctly', () => {
  const component = renderer.create(
    <MuiThemeProvider muiTheme={getMuiTheme(commuteTheme)}>
      <LoginPanel />
    </MuiThemeProvider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
