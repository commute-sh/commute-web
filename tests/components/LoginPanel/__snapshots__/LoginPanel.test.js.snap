exports[`test LoginPanel renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
    }
  }>
  <div
    className={undefined}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#ffffff",
        "borderRadius": "2px",
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "margin": 12,
        "minWidth": 88,
        "muiPrepared": true,
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      }
    }>
    <a
      disabled={false}
      href="https://facebook.com"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "#ff4081",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transform": "translate(0, 0)",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": "middle",
          "width": "100%",
        }
      }
      tabIndex={0}
      target="_blank"
      type="button">
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}>
        <div
          style={
            Object {
              "backgroundColor": false,
              "borderRadius": 2,
              "height": 36,
              "muiPrepared": true,
              "top": 0,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }>
          <span
            className="muidocs-icon-custom-facebook"
            color="#ffffff"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "#ffffff",
                "display": "inline-block",
                "fontSize": 24,
                "marginLeft": 12,
                "marginRight": 0,
                "muiPrepared": true,
                "position": "relative",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "verticalAlign": "middle",
              }
            } />
          <span
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "margin": 0,
                "muiPrepared": true,
                "opacity": 1,
                "paddingLeft": 8,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "userSelect": "none",
              }
            }>
            Facebook Login
          </span>
        </div>
      </div>
    </a>
  </div>
  <div
    className={undefined}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#ffffff",
        "borderRadius": "2px",
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "margin": 12,
        "minWidth": 88,
        "muiPrepared": true,
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      }
    }>
    <a
      disabled={false}
      href="https://google.com"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "#ff4081",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transform": "translate(0, 0)",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": "middle",
          "width": "100%",
        }
      }
      tabIndex={0}
      target="_blank"
      type="button">
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}>
        <div
          style={
            Object {
              "backgroundColor": false,
              "borderRadius": 2,
              "height": 36,
              "muiPrepared": true,
              "top": 0,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }>
          <span
            className="muidocs-icon-custom-google"
            color="#ffffff"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "#ffffff",
                "display": "inline-block",
                "fontSize": 24,
                "marginLeft": 12,
                "marginRight": 0,
                "muiPrepared": true,
                "position": "relative",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "verticalAlign": "middle",
              }
            } />
          <span
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "margin": 0,
                "muiPrepared": true,
                "opacity": 1,
                "paddingLeft": 8,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "userSelect": "none",
              }
            }>
            Google Login
          </span>
        </div>
      </div>
    </a>
  </div>
  <div
    className={undefined}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#ffffff",
        "borderRadius": "2px",
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "margin": 12,
        "minWidth": 88,
        "muiPrepared": true,
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      }
    }>
    <a
      disabled={false}
      href="https://twitter.com"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "#ff4081",
          "border": 10,
          "borderRadius": 2,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "position": "relative",
          "textAlign": "center",
          "textDecoration": "none",
          "transform": "translate(0, 0)",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": "middle",
          "width": "100%",
        }
      }
      tabIndex={0}
      target="_blank"
      type="button">
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}>
        <div
          style={
            Object {
              "backgroundColor": false,
              "borderRadius": 2,
              "height": 36,
              "muiPrepared": true,
              "top": 0,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            }
          }>
          <span
            className="muidocs-icon-custom-twitter"
            color="#ffffff"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "#ffffff",
                "display": "inline-block",
                "fontSize": 24,
                "marginLeft": 12,
                "marginRight": 0,
                "muiPrepared": true,
                "position": "relative",
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "verticalAlign": "middle",
              }
            } />
          <span
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 14,
                "fontWeight": 500,
                "letterSpacing": 0,
                "margin": 0,
                "muiPrepared": true,
                "opacity": 1,
                "paddingLeft": 8,
                "paddingRight": 16,
                "position": "relative",
                "textTransform": "uppercase",
                "userSelect": "none",
              }
            }>
            Twitter Login
          </span>
        </div>
      </div>
    </a>
  </div>
</div>
`;
